art.hpp

```diff
@@ -48,32 +56,69 @@ using node_ptr = basic_node_ptr<node_header>;
 
 struct impl_helpers;
 
-using inode_defs = unodb::detail::basic_inode_def<inode, inode_4, inode_16,
-                                                  inode_48, inode_256>;
+template <typename Key>
+using inode_defs = basic_inode_def<inode<Key>, inode_4<Key>, inode_16<Key>,
+                                   inode_48<Key>, inode_256<Key>>;
 
-template <class INode>
-using db_inode_deleter =
-    unodb::detail::basic_db_inode_deleter<INode, unodb::db>;
+template <typename Key, class INode>
+using db_inode_deleter = basic_db_inode_deleter<Key, INode, unodb::db>;
 
-using art_policy = unodb::detail::basic_art_policy<
-    unodb::db, unodb::in_fake_critical_section, unodb::fake_lock,
-    unodb::fake_read_critical_section, unodb::detail::node_ptr, inode_defs,
-    db_inode_deleter, unodb::detail::basic_db_leaf_deleter>;
+template <typename Key>
+using art_policy =
+    basic_art_policy<Key, unodb::db, unodb::in_fake_critical_section,
+                     unodb::fake_lock, unodb::fake_read_critical_section,
+                     node_ptr, inode_defs, db_inode_deleter,
+                     basic_db_leaf_deleter>;
 
-using inode_base = unodb::detail::basic_inode_impl<art_policy>;
+template <typename Key>
```

Comment:
@laurynas-biveinis We should be brave and also put in a template argument for the Value. We can constrain it to always be span for now (existing code), but then open it up to other strongly typed keys later.